plugins/aggregate/aggregate.cpp

```diff
@@ -422,9 +421,12 @@ struct aggregation {
       auto result = std::vector<bool>{};
       result.resize(builder_->num_fields(), false);
       if (!buckets_.empty())
-        for (int column = 0; column < builder_->num_fields(); ++column)
-          result[column] = !builder_->GetField(column)->type()->Equals(
-            buckets_.begin()->second[column]->type);
+        for (int column = 0; column < builder_->num_fields(); ++column) {
+          auto scalar = buckets_.begin()->second[column];
+          result[column]
+            = scalar
+              && !builder_->GetField(column)->type()->Equals(scalar->type);
```

Comment:
With this change, can we remove the `&& scalar condition` in line 437 below?
```
if (column_needs_cast[column] && scalar) {
```

Reply:
When `column_needs_cast[column]` evaluates to `true` then it is still needed. On 425 the check is only done for the first field (`buckets_.begin()`), the check on 437 is done for each field.

Reply:
I have changed this behavior.